platform/macos/detect.py

```diff
@@ -86,27 +84,15 @@ def configure(env):
 
     ## Build type
 
-    if env["target"] == "release":
-        if env["optimize"] == "speed":  # optimize for speed (default)
-            env.Prepend(CCFLAGS=["-O3", "-fomit-frame-pointer", "-ftree-vectorize"])
-        elif env["optimize"] == "size":  # optimize for size
-            env.Prepend(CCFLAGS=["-Os", "-ftree-vectorize"])
+    # TODO: Re-evaluate the need for this / streamline with common config.
+    if env["target"] == "template-release":
+        if env["optimize"].startswith("speed"):
+            env.Prepend(CCFLAGS=["-fomit-frame-pointer", "-ftree-vectorize"])
+        elif env["optimize"] == "size":
+            env.Prepend(CCFLAGS=["-ftree-vectorize"])
         if env["arch"] != "arm64":
             env.Prepend(CCFLAGS=["-msse2"])
```

Comment:
Our handling of `SSE2` is a bit arbitrary. We seem to forcefully define it for Windows MinGW and macOS x86_64 with `target=release` specifically, and not on any other platform:
```
$ rg msse2
platform/windows/detect.py
496:        env.Append(CCFLAGS=["-msse2"])

platform/macos/detect.py
95:            env.Prepend(CCFLAGS=["-msse2"])

modules/raycast/SCsub
70:            env_raycast.Append(CPPFLAGS=["-msse2", "-mxsave"])
```